components/expenses/ExpensesPage.tsx

```diff
@@ -1,158 +1,273 @@
+    scheduledExpenses: expenses(
+      host: $account
+      status: SCHEDULED_FOR_PAYMENT
+      payoutMethodType: BANK_ACCOUNT
+      limit: 1
+    ) {
+      totalCount
+    }
+  }
 
-  const hasFilters = React.useMemo(() => {
-    return Object.entries(query).some(([key, value]) => key !== 'offset' && key !== 'limit' && value);
-  }, [query]);
+  ${expensesListFieldsFragment}
+  ${accountNavbarFieldsFragment}
+  ${expenseHostFields}
+`;
 
-  function buildFilterLinkParams(params) {
-    const queryParameters = {
-      ...omit(query, ['offset', 'collectiveSlug', 'parentCollectiveSlug']),
-      ...params,
-    };
+export const schema = commonSchema.extend({ direction: expenseDirectionFilter.schema });
 
-    return omitBy(queryParameters, value => !value);
-  }
+type FilterValues = z.infer<typeof schema>;
 
-  function updateFilters(queryParams) {
-    return router.push({
-      pathname: expensesRoute,
-      query: buildFilterLinkParams({ ...queryParams, offset: null }),
-    });
-  }
+type FilterMeta = CommonFilterMeta & {
+  expenseTags?: string[];
+  includeUncategorized?: boolean;
+};
 
-  function handleSearch(searchTerm) {
-    const params = buildFilterLinkParams({ searchTerm, offset: null });
-    router.push({ pathname: expensesRoute, query: params });
-  }
+export const toVariables: FiltersToVariables<FilterValues, HostDashboardExpensesQueryVariables, FilterMeta> = {
+  ...commonToVariables,
+  limit: (value, key) => ({ [key]: value * 2 }), // Times two for the lazy pagination
+};
 
-  function getTagProps(tag) {
-    if (tag === query.tag) {
-      return { type: 'info', closeButtonProps: true };
+const filters: FilterComponentConfigs<FilterValues, FilterMeta> = {
+  ...pick(commonFilters, ['searchTerm']),
+  direction: expenseDirectionFilter.filter,
+  ...omit(commonFilters, ['searchTerm', 'status']),
+  status: { ...commonFilters.status, static: false } as typeof commonFilters.status,
+  tag: expenseTagFilter.filter,
+};
+
+type ExpensesProps = {
+  account: ExpensesPageQuery['account'];
+  expenses: ExpensesPageQuery['expenses'];
+  direction: EXPENSE_DIRECTION;
+};
+
+const Expenses = ({ account, expenses: _expenses, direction }: ExpensesProps) => {
+  const router = useRouter();
+  const isSubmitted = direction === EXPENSE_DIRECTION.SUBMITTED;
+  const { LoggedInUser } = useLoggedInUser();
+  const meta: FilterMeta = {
+    currency: account?.currency,
+    expenseTags: account?.expensesTags?.map(tag => tag.tag) || [],
+  };
+
+  const queryFilter = useQueryFilter({
+    schema,
+    toVariables,
+    filters: isSubmitted ? omit(filters, ['direction']) : filters,
+    defaultFilterValues: { direction: direction || EXPENSE_DIRECTION.RECEIVED },
+    meta,
+    shallow: true,
+  });
+
+  const variables = {
+    ...{
+      [queryFilter.values.direction === EXPENSE_DIRECTION.RECEIVED ? 'account' : 'fromAccount']: {
+        legacyId: account?.legacyId,
+      },
+    },
+    accountSlug: account?.slug,
```

Comment:
Here's a fix to include these in the general `toVariables` function, so that all variables can be created from the query filter toVariables function which is used when creating the SSR variables:

https://github.com/opencollective/opencollective-frontend/pull/11286 (targeting this branch)